src/ring_theory/ideal/basic.lean

```diff
@@ -256,6 +256,31 @@ begin
   rwa ← span_eq I,
 end
 
+lemma has_left_inv_iff_span_top {x : α} :
+  has_left_inv x ↔ span ({x} : set α) = ⊤ :=
+begin
+  split,
+  { rintro ⟨a, hax⟩,
+    apply eq_top_of_unit_mem _ x a _ hax,
+    apply submodule.mem_span_singleton_self },
+  { intro h,
+    have : (1 : α) ∈ span ({x} : set α) := by { rw h, exact submodule.mem_top },
+    exact (mem_span_singleton').mp this },
+end
+
+lemma not_has_left_inv_iff_mem_maximal {x : α} :
```

Comment:
This looks related to `exists_max_ideal_of_mem_nonunits`, right? Can you strengthen that lemma to an iff?

Reply:
Yes right, thanks!
I replaced `exists_max_ideal_of_mem_nonunits` with `mem_nonunits_iff_exists_max_ideal`.